### ISSUE TITLE
How to disable withContributions for models like GLM after it was set in H2OAutoML

### ISSUE THREAD

**Issue by dzlab** [h2oai/sparkling-water#2448]
I'm training an automl for a binary classification problem with the following options, note that i'm enabling `withContributions` parameter
```scala
import ai.h2o.sparkling.ml.algos.H2OAutoML
val automl = new H2OAutoML()
      .setFeaturesCols(params.featureCols)
      .setLabelCol(params.labelCol)
      .setColumnsToCategorical(params.categoricalCols)
      .setSplitRatio(params.splitRatio)
      .setWithDetailedPredictionCol(true)
      .setWithContributions(params.contributions)
      .setNfolds(params.nFolds)
      .setSeed(params.seed)
      .setWeightCol(params.weightCol)
      .setMaxModels(autoMLParams.maxModels)
      .setClassSamplingFactors(autoMLParams.classSamplingFactors)
      .setConvertInvalidNumbersToNa(autoMLParams.convertInvalidNumbersToNa)
      .setExcludeAlgos(autoMLParams.excludeAlgos)
      .setFoldCol(autoMLParams.foldCol)
      .setIgnoredCols(autoMLParams.ignoredCols)
      .setMaxAfterBalanceSize(autoMLParams.maxAfterBalanceClasses)
      .setMaxRuntimeSecs(autoMLParams.maxRuntimeSecs)
      .setSortMetric(autoMLParams.sortMetric)
      .setStoppingMetric(autoMLParams.stoppingMetric)
      .setStoppingRounds(autoMLParams.stoppingRounds)
      .setStoppingTolerance(autoMLParams.stoppingTolerance)
```

One of the models in the leaderboard is a GLM model, when I try to use for predictions with `model.transform(testDF)` I get
```
Caused by: java.lang.IllegalArgumentException: Computing contributions is not allowed on MOJO of type 'class hex.genmodel.algos.glm.GlmMojoModel'!
	at ai.h2o.sparkling.ml.models.H2OMOJOCache$.canGenerateContributions(H2OMOJOModel.scala:318)
	at ai.h2o.sparkling.ml.models.H2OMOJOCache$.loadMojoBackend(H2OMOJOModel.scala:345)
	at ai.h2o.sparkling.ml.models.H2OMOJOCache$.loadMojoBackend(H2OMOJOModel.scala:304)
	at ai.h2o.sparkling.ml.models.H2OMOJOBaseCache$class.getMojoBackend(H2OMOJOBaseCache.scala:78)
	at ai.h2o.sparkling.ml.models.H2OMOJOCache$.getMojoBackend(H2OMOJOModel.scala:304)
	at ai.h2o.sparkling.ml.models.H2OMOJOPrediction$class.getPredictionUDF(H2OMOJOPrediction.scala:54)
	at ai.h2o.sparkling.ml.models.H2OMOJOModel.getPredictionUDF(H2OMOJOModel.scala:42)
	at ai.h2o.sparkling.ml.models.H2OMOJOModel$$anonfun$2.apply(H2OMOJOModel.scala:105)
	at ai.h2o.sparkling.ml.models.H2OMOJOModel$$anonfun$2.apply(H2OMOJOModel.scala:105)
	at ai.h2o.sparkling.ml.models.H2OSupervisedMOJOModel.applyPredictionUdfToFlatDataFrame(H2OSupervisedMOJOModel.scala:51)
	at ai.h2o.sparkling.ml.models.H2OMOJOFlattenedInput$class.applyPredictionUdf(H2OMOJOFlattenedInput.scala:40)
	at ai.h2o.sparkling.ml.models.H2OMOJOModelBase.applyPredictionUdf(H2OMOJOModelBase.scala:25)
	at ai.h2o.sparkling.ml.models.H2OMOJOModel.transform(H2OMOJOModel.scala:105)
```

So I use this hack to disable `withContributions` and then calling transform successfully:
```scala
   val model: H2OMOJOModel = ...
    val withContributions = model.getParam("withContributions")
    model.set(withContributions, false)
```

Now if I persist the MOJO model and load it again into a H2OMOJOModel, and I call `model.transform` again I hit this error
```
Caused by: java.lang.IllegalStateException: Contributions were not enabled using in EasyPredictModelWrapper (use setEnableContributions).
	at hex.genmodel.easy.EasyPredictModelWrapper.getContributionNames(EasyPredictModelWrapper.java:390)
	at ai.h2o.sparkling.ml.models.PredictionWithContributions$class.getContributionsSchema(PredictionWithContributions.scala:25)
	at ai.h2o.sparkling.ml.models.H2OMOJOModel.getContributionsSchema(H2OMOJOModel.scala:42)
	at ai.h2o.sparkling.ml.models.H2OMOJOPredictionBinomial$class.getBinomialPredictionSchema(H2OMOJOPredictionBinomial.scala:84)
	at ai.h2o.sparkling.ml.models.H2OMOJOModel.getBinomialPredictionSchema(H2OMOJOModel.scala:42)
	at ai.h2o.sparkling.ml.models.H2OMOJOPredictionBinomial$class.getBinomialPredictionUDF(H2OMOJOPredictionBinomial.scala:41)
	at ai.h2o.sparkling.ml.models.H2OMOJOModel.getBinomialPredictionUDF(H2OMOJOModel.scala:42)
	at ai.h2o.sparkling.ml.models.H2OMOJOPrediction$class.getPredictionUDF(H2OMOJOPrediction.scala:56)
	at ai.h2o.sparkling.ml.models.H2OMOJOModel.getPredictionUDF(H2OMOJOModel.scala:42)
	at ai.h2o.sparkling.ml.models.H2OMOJOModel$$anonfun$2.apply(H2OMOJOModel.scala:105)
	at ai.h2o.sparkling.ml.models.H2OMOJOModel$$anonfun$2.apply(H2OMOJOModel.scala:105)
	at ai.h2o.sparkling.ml.models.H2OSupervisedMOJOModel.applyPredictionUdfToFlatDataFrame(H2OSupervisedMOJOModel.scala:51)
	at ai.h2o.sparkling.ml.models.H2OMOJOFlattenedInput$class.applyPredictionUdf(H2OMOJOFlattenedInput.scala:40)
	at ai.h2o.sparkling.ml.models.H2OMOJOModelBase.applyPredictionUdf(H2OMOJOModelBase.scala:25)
	at ai.h2o.sparkling.ml.models.H2OMOJOModel.transform(H2OMOJOModel.scala:105)
```

What is the best way to reset the `withContributions` flag in case the GLM model is part of the automl leaderboard.

**Comment by mn-mikke** [comment]
> One of the models in the leaderboard is a GLM model, when I try to use for predictions with model.transform(testDF) I get an exception

This is a bug, thanks for reporting it! You can track progress of the bug fix [here](https://h2oai.atlassian.net/browse/SW-2528).

> Now if I persist the MOJO model and load it again into a H2OMOJOModel, and I call model.transform again I hit this error

@dzlab How do you save and load the model? Can you provide an example?

**Comment by dzlab** [comment]
@mn-mikke this is how I save/load model
```
  val baseURL: String = ...
  def saveMOJOModel(model: H2OMOJOModel): String = {
    val id = UUID.randomUUID().toString
    val modelPath = s"$baseURL/$id"
    logger.info(s"Saving model to $modelPath")
    model.save(modelPath)
    id
  }

  def loadModel(id: String): H2OMOJOModel = {
    val modelPath = s"$baseURL/$id"
    val reader = new ModelReader[H2OMOJOModel]
    reader.load(modelPath)
  }
```
What is the proper way for setting the `withContributions` param before running a prediction with the model?

**Comment by mn-mikke** [comment]
Hi @dzlab,
[this problem](https://h2oai.atlassian.net/browse/SW-2528) will be fixed in the next SW version `3.32.1.1-1`

After disabling contributions on the model:
```
   val model: H2OMOJOModel = ...
    val withContributions = model.getParam("withContributions")
    model.set(withContributions, false)
```

I was able to save, load and score with the model. I just had to change your `loadModel` method to:
```
    def loadModel(id: String): H2OMOJOModel = {
      val modelPath = s"$baseURL/$id"
      H2OMOJOModel.load(modelPath)
    }
```

**Comment by dzlab** [comment]
Great thanks a lot for the quick fix.
